nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -1433,6 +1433,14 @@ public void synchronize(final ParameterContext parameterContext, final Versioned
                 } else {
                     final Map<String, Parameter> updatedParameters = createParameterMap(proposed.getParameters());
 
+                    // If any parameters are removed, need to add a null value to the map in order to make sure that the parameter is removed.
```

Comment:
It looks like this is also a problem in `synchronize(final ProcessGroup group, final VersionedProcessGroup proposed, final Map<String, VersionedParameterContext> versionedParameterContexts)`, which is the code path used to synchronize a versioned group.  What do you think about adding similar logic at line 1962?

Reply:
Good point. The synchronize for the whole process group is primarily used when changing version with NiFi Registry, and in that case we don't want to remove parameters because it isn't clear if the parameter context is used by multiple version controlled flows. 

For the case in this PR, it is the specific synchronize method for a parameter context where some external system is specifying that the desired state is for the parameter context. So I think it makes sense to only add this here.